### ISSUE TITLE
dbdeployer sandbox setup fails with MariaDB 10.4.3

### ISSUE THREAD

**Issue by dveeden** [datacharmer/dbdeployer#67]
In MariaDB 10.4.3 socket authentication is used by default.
This causes the sandbox setup to fail.
https://mariadb.com/kb/en/library/authentication-plugin-unix-socket/

I plan to work on a fix for this.

**Comment by datacharmer** [comment]
Thanks for reporting.
Before submitting a fix, would you care discussing here what method you would implement?
Notice that with the recent addition of "capabilities", it becomes easier to add solutions customized for a given flavor.

**Comment by dveeden** [comment]
This is working:
```dbdeployer deploy single 10.4.3 --init-options '--auth-root-authentication-method=normal'```

**Comment by datacharmer** [comment]
> This is working:
> `dbdeployer deploy single 10.4.3 --init-options '--auth-root-authentication-method=normal'`

Only if you are running as root.
As a normal user, even when changing the init options, you get:
```
chown: changing ownership of '/home/gmax/opt/mysql/ma10.4.3/lib/plugin/auth_pam_tool_dir/auth_pam_tool': Operation not permitted
Couldn't set an owner to '/home/gmax/opt/mysql/ma10.4.3/lib/plugin/auth_pam_tool_dir/auth_pam_tool'.
 It must be root, the PAM authentication plugin doesn't work otherwise..
```
This is primarily a MariaDB bug.

**Comment by dveeden** [comment]
I didn't get the error for chown. I'm running as a normal (non-root) user.

**Comment by datacharmer** [comment]
I got the error the first time I tried. 
After I tried as a root user, the next attempts succeeded. I wonder what the problem is.

**Comment by datacharmer** [comment]
Fixed in release v1.22.0
